File: 858-mirror-reflection/858-mirror-reflection.java
class Solution {
    public int mirrorReflection(int p, int q) {
        int e=q,r=p;
        while(e%2==0 && r%2==0){
            e/=2;
            r/=2;
        }
        if(e%2==0 && r%2!=0)  return 0;
        else if(e%2!=0 && r%2==0)  return 2;
        else  return 1;
    }
}

File: 858-mirror-reflection/858-mirror-reflection.py
import math
class Solution:
    def mirrorReflection(self, p: int, q: int) -> int:
        e,r = q,p
        while(e%2==0 and r%2==0):
            e//=2
            r//=2
        if(e%2==0 and r%2!=0):
            return 0
        if(e%2!=0 and r%2==0):
            return 2
        if(e%2!=0 and r%2!=0):
            return 1